src/EthPool.sol

```diff
@@ -77,6 +77,21 @@ contract EthPool is Pool {
             );
     }
 
+    /// @notice Gives funds from the sender of the message to the receiver.
+    /// The receiver can collect them immediately.
+    /// @param receiver The receiver
+    function give(address receiver) public payable {
+        _giveInternal(msg.sender, receiver, uint128(msg.value));
+    }
+
+    /// @notice Gives funds from the sub-sender of the sender of the message to the receiver.
+    /// The receiver can collect them immediately.
+    /// @param subSenderId The id of the giver's sub-sender
+    /// @param receiver The receiver
+    function giveFromSubSender(uint256 subSenderId, address receiver) public payable {
+        _giveFromSubSenderInternal(msg.sender, subSenderId, receiver, uint128(msg.value));
+    }
+
     function _transfer(address userAddr, int128 amt) internal override {
```

Comment:
In the give case of an ETH pool. we are actually done and the checks and sending back are not required.

We could make a method `giveTransfer` which for the erc20 case just calls `transfer` and is empty for the ETH pool but I guess its more logic and not really worth it.

Reply:
Yeah, it's stupid that the ETH pool does that step now. But it simplifies logic at a cost of two arithmetic operations.